hathor/transaction/storage/transaction_storage.py

```diff
@@ -753,15 +1067,23 @@ def _topological_sort_dfs(self, root: BaseTransaction, visited: Dict[bytes, int]
             # matter.
             for parent_hash in tx.parents[::-1]:
                 if parent_hash not in visited:
-                    parent = self.get_transaction(parent_hash)
-                    stack.append(parent)
+                    try:
+                        parent = self.get_transaction(parent_hash)
+                    except TransactionDoesNotExist:
+                        pass
+                    else:
+                        stack.append(parent)
 
             for txin in tx.inputs:
                 if txin.tx_id not in visited:
-                    txinput = self.get_transaction(txin.tx_id)
-                    stack.append(txinput)
-
-    def _add_to_cache(self, tx: BaseTransaction) -> None:
+                    try:
+                        txinput = self.get_transaction(txin.tx_id)
+                    except TransactionDoesNotExist:
+                        pass
```

Comment:
Maybe add a comment explaining that it might fail because some dependencies might not exist.

Reply:
Done.